components/invoice.js

```diff
@@ -117,28 +110,37 @@ const MutationInvoice = ({ id, hash, hmac, errorCount, repeat, onClose, expiresA
     return <QrSkeleton description status='loading' />
   }
 
+  const retry = !!onRetry
   let errorStatus = 'Something went wrong trying to perform the action after payment.'
-  if (errorCount > 1) {
+  if (retryError > 0) {
     errorStatus = 'Something still went wrong.\nYou can retry or cancel the invoice to return your funds.'
   }
 
   return (
     <>
-      <Invoice invoice={data.invoice} modal {...props} />
-      {errorCount > 0
+      <Invoice invoice={data.invoice} modal onPayment={onPayment} successVerb='received' />
+      {retry
         ? (
           <>
             <div className='my-3'>
               <InvoiceStatus variant='failed' status={errorStatus} />
             </div>
             <div className='d-flex flex-row mt-3 justify-content-center'>
-              <Button className='mx-1' variant='info' onClick={repeat}>Retry</Button>
               <Button
-                className='mx-1'
-                variant='danger' onClick={async () => {
-                  await cancelInvoice({ variables: { hash, hmac } })
-                  onClose()
+                className='mx-1' variant='info' onClick={async () => {
+                  try {
+                    await onRetry()
+                  } catch (err) {
+                    console.error('retry error:', err)
+                    setRetryError(retryError + 1)
```

Comment:
nit: if you're referencing prior state, you usually want to use the function form of `setState`:

`setRetryError(retryError => retryError + 1)`